### PR TITLE
Rf 1.0.4.1

### PR DESCRIPTION
**Pull request by asa1997**
- Steps for manual release added in Readme.md.
- Files added, release.sh and dopublish.sh.
-Template file added.
-Automata files deleted from /bin.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,13 @@ Other useful commands:
         kob version     
          
 
+### Manual Release
```
Comment (panickervinod) [inline README.md:98]: Move instructions to  document maintainers-handbook.md

Have  section preparing for Release.

File: bin/dopublish.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+kob_rel_version=$1
+kob_archive_download_repo="KOBman_target_repo"
+branch="REL-$kob_rel_version"
+kob_namespace= ${KOB_NAMESPACE:-hyperledgerkochi}
+#Checkout latest tag
+# The branch is used to make the tar files.
+git branch -D $kob_rel_version-branch
```
Comment (panickervinod) [inline bin/dopublish.sh:9]: Branch creation should be done in release. 

You need to only check if the branch is available , if mention nothing is available to publish.

File: bin/release.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    then
+        echo "Usage: release.sh <version>"
+ 	    exit 0
+fi
+
+echo "$kob_version"
+#prepare branch
+# echo "prepare branch"
+# #git fetch
+git checkout Dev
```
Comment (panickervinod) [inline bin/release.sh:22]: dev should be Small caps

File: bin/release.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+kob_version="$1"
+
+branch="REL-taggin"
```
Comment (panickervinod) [inline bin/release.sh:5]: branch should be REL-${kob_version}

File: bin/release.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+#prepare branch
+# echo "prepare branch"
+# #git fetch
+git checkout Dev
+git branch -D "$branch"
+git checkout -b "$branch"
+
+#copy the tmpl file to /scripts
+echo "copying tmpl file to scripts"
+cp /$HOME/$USR/KOBman/scripts/tmpl/get.kobman.io.tmpl /$HOME/$USR/KOBman/scripts/
```
Comment (panickervinod) [inline bin/release.sh:28]: This should work for automated as well as manual, validate the location

File: scripts/tmpl/get.kobman.io.tmpl [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+echo "Attempt update of zsh profile..."
+touch "$kobman_zshrc"
+if [[ -z $(grep 'kobman-init.sh' "$kobman_zshrc") ]]; then
+    echo -e "\n$kobman_init_snippet" >> "$kobman_zshrc"
+    echo "Updated existing ${kobman_zshrc}"
+fi
+
+sudo chmod a+rwx .
+sudo chmod u+xr /home/blockchain/.kobman/candidates
```
Comment (panickervinod) [inline scripts/tmpl/get.kobman.io.tmpl:247]: Remove Hard coding.

File: scripts/tmpl/get.kobman.io.tmpl [hunk 7]
```diff
@@ -0,0 +1,262 @@
+
+#!/bin/bash
+
+#Install: stable
+
+# Global variables
+KOBMAN_VERSION="@KOB_VERSION@"
+KOB_ARCHIVE_DOWLOAD_REPO="@KOB_ARCHIVE_DOWLOAD_REPO@"
```
Comment (panickervinod) [inline scripts/tmpl/get.kobman.io.tmpl:8]: Check if KOB or KOBMAN_VERSION, Please make it consistent.

File: scripts/tmpl/get.kobman.io.tmpl [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+KOBMAN_PLATFORM=$(uname)
+KOBMAN_SERVICE="https://raw.githubusercontent.com"
```
Comment (panickervinod) [inline scripts/tmpl/get.kobman.io.tmpl:10]: Mention what service this is based on end point. This must point to the gh-pages and should have the namespace

File: scripts/tmpl/get.kobman.io.tmpl [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+# KOBMAN_NAMESPACE="EtricKombat"
```
Comment (panickervinod) [inline scripts/tmpl/get.kobman.io.tmpl:11]: remove hard coding , make use of export to set KOBMAN_NAMESPACE

**Review by panickervinod — CHANGES_REQUESTED** [review]

**Comment by panickervinod** [comment]
Please incorporate the review comments. 

a) Release branch creation to be done in release.sh not in dopublish.
b) Add checks to find if the release branch already exists.
b) Have a wrapper function for all echo, follow sdkman code for the same. release.sh will be used for both manual as well as automated release prepartion. 
c) Review tmpl code, 
d) set export variable for KOB_NAMESPACE. Remove multiple hardcoding in tmpl

File: README.md (hunk 1 above)
Comment (asa1997) [reply README.md:98]: maintainers-handbook.md
     only the instructions for manual release??
Have section preparing for Release.
     Please do guide me on this

Comment (panickervinod) [reply README.md:98]: Have a document for Maintainers , doing a release will be one activity for them .

Move this under that.

File: bin/release.sh (hunk 3 above)
Comment (asa1997) [reply bin/release.sh:22]: This Dev is the name of my remote branch. I have been working from it

**Pull request closed without merging**